tree/ntuplebrowse/v7/inc/ROOT/RNTupleBrowser.hxx

```diff
@@ -0,0 +1,201 @@
+   /// Keeps a list of previous directories. When a previously used directory appears again, the necessary fReaderPtr can be directly assigned from fReaderPtrVec instead of doing costly operations and creating a new one.
+   std::vector<TDirectory*>                     fDirectoryVec;
+   /// Holds the currently used RNTupleReader. It is changed when the folder directly beneath the .root file is double-clicked.
+   std::shared_ptr<RNTupleReader>               fReaderPtr;
+   // holds previously used RNTupleReader pointers. This allows the destructor for RNTupleReader to be called when the destructor for RNTupleBrowser is called. The problems when deleting a RNTupleReader earlier are:
+   // 1. deleting RNTupleReader causes a chain reaction of destructors being called, which even deletes the currently displayed TH1F Histo. Instead of modifying code on other places which could influence other parts of the program, the problem was fixed here.
+   // 2. Coming back to a field in rntuple after a field from a different ntuple and not calling SetDirectory(TDirectory*) results in segmentation fault, if the associated RNTupleReader is deleted.
+   /// Stores pointers of created RNTupleReaders so they can be deleted in ~RNTupleBrowser.
+   std::vector<std::shared_ptr<RNTupleReader>>  fReaderPtrVec;
+   // For normal use the value is always set to 0. In case of a unit test, the value is >0 and can give feedback to the unittest if something worked or not. For details, see test/ntuplebrowse.cxx
+   /// Used for testing. Should always be 0 during normal use.
+   int                                          fUnitTest = 0;
+   
+public:
+   // Allows to keep created TH1F histo outside created function. Else user gets to see the histogram for less than a second. Only points to histograms created through RNTupleElementField.
+   /// Points to currently displayed histogram.
+   TH1F*                                        fCurrentTH1F;
+   // Keeps hold of pointers of dynamically allocated objects representing rntuple-fields in. They are all deleted in one step in the destructor. Their job is to prevent memory leak. Deleting these object before calling the destructor would make them uncklickable in TBrowser.
+   /// Stores instances of RNTupleFieldElement and RNTupleFieldElementFolder to prevent memory leak.
+   std::vector<TNamed*>                         fNTupleBrowsePtrVec;
+   
+   
+   RNTupleBrowser(TDirectory* directory=nullptr, int fUnitTest = 0);
+   ~RNTupleBrowser();
+   
+   /// Instantiates the RNTupleReader associated to the TDirectory.
+   void SetDirectory(TDirectory* directory);
+   /// Called when double-clicking the folder which appears afer double-clicking the .root file.
+   void Browse(TBrowser *b);
+   RNTupleReader* GetReaderPtr() const { return fReaderPtr.get(); }
+   int GetfUnitTest() const { return fUnitTest; }
+   void IncreasefUnitTest() { ++fUnitTest; }
+   Bool_t IsFolder() const { return kTRUE; }
+   
+   ClassDef(RNTupleBrowser,2)
```

Comment:
The ClassDef for this class was removed (and inheritance from TNamed too), because it interfered with creating unique-pointers. For the other classes I set it to zero.